mitodl/edx-platform
# Reimplemented legacy remote gradebook merge/overload functionality

#### What are the relevant tickets?

Closes #14
⚠️  REBASED ON #19 ⚠️ 

#### What's this PR do?

Reimplements remote gradebook merge/overload functionality (shown in this issue: https://github.com/mitocw/edx-platform/issues/76; live example: https://staging.mitx.mit.edu/courses/DevOps/0.001/2015_Summer/legacy_instructor_dash)

#### Where should the reviewer start?

Probably `lms/templates/instructor/instructor_dashboard_2/remote_gradebook.html` and  `lms/static/js/instructor_dashboard/remote_gradebook.js`

#### How should this be manually tested?

- Make sure you have staff access for your test course
- Set the `ENABLE_INSTRUCTOR_REMOTE_GRADEBOOK_CONTROLS` feature to `true` (in `lms.env.json`)
- Create a `REMOTE_GRADEBOOK` setting in `lms.env.json`. This should be a dict with 2 values: 
    ```
    "REMOTE_GRADEBOOK": {
        "URL": "http://lmodproxyqa-mitodl.rhcloud.com/edx_grades",
        "DEFAULT_NAME": "STELLAR:/project/mitxdemosite"
    }
    ```

- Create `REMOTE_GRADEBOOK_USER` and `REMOTE_GRADEBOOK_PASSWORD` settings in `lms.auth.json`. Ask me or Peter for these values
- Navigate to the `Remote Gradebook` tab for some course (eg: `http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-remote_gradebook`)
- Click the "Merge"/"Overload enrollment list using remote gradebook" buttons and make sure the results are as expected. "Merge" is intended to take all users for a specific section in the remote gradebook and enroll them in the given course in edX. "Overload" does the same thing, but also unenrolls all currently enrolled non-staff users for the given course. If you have such users in your local environment, you should be prompted to confirm that you want to unenroll those users
